src/main/java/org/sqlite/util/OSInfo.java

```diff
@@ -171,9 +171,11 @@ static String resolveArmArchType() {
             // armType (uname -m) can be armv5t, armv5te, armv5tej, armv5tejl, armv6, armv7, armv7l,
             // aarch64, i686
 
+            boolean is32bit = "32".equals(System.getProperty("sun.arch.data.model"));
+
             // for Android, we fold everything that is not aarch64 into arm
             if (isAndroid()) {
-                if (armType.startsWith("aarch64")) {
+                if (armType.startsWith("aarch64") && !is32bit) {
```

Comment:
i don't think we need the change in Android, not even sure those system properties would exist on Android